File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/Deduction.cs
namespace HRManagementSystem.Domain.Entities;

public class Deduction: BaseEntity
{
    public string Name { get; set; } = default!;
    public decimal Amount { get; set; }
     public bool IsPercentage { get; set; } = false; 
    public ICollection<EmployeeDeduction> EmployeeDeductions { get; set; } = new List<EmployeeDeduction>();

}

File: frontend/src/app/models/deduction.ts
export interface DeductionDto {
    id: number;
    name: string;
    amount: number;
    isPercentage: boolean;
}

export interface CreateDeductionDto {
    name: string;
    amount: number;
    isPercentage: boolean;
}

export interface UpdateDeductionDto {
    name: string;
    amount: number;
    isPercentage: boolean;
}
